src/platform/unittest/ToeplitzTest.cpp

```diff
@@ -0,0 +1,182 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+--*/
+
+#define _CRT_SECURE_NO_WARNINGS 1
+#include "main.h"
+#include "msquic.h"
+#include "msquichelper.h"
+#include "quic_toeplitz.h"
+#include <stdio.h>
+
+#ifdef QUIC_CLOG
+#include "ToeplitzTest.cpp.clog.h"
+#endif
+
+struct ToeplitzTest : public ::testing::Test
+{
+    protected:
+    static const char* HashKey;
+
+    struct QuicBuffer
+    {
+        uint8_t* Data;
+        uint16_t Length;
+
+        QuicBuffer(const char* HexBytes)
+        {
+            Length = (uint16_t)(strlen(HexBytes) / 2);
+            Data = new uint8_t[Length];
+
+            for (uint16_t i = 0; i < Length; ++i) {
+                Data[i] =
+                    (DecodeHexChar(HexBytes[i * 2]) << 4) |
+                    DecodeHexChar(HexBytes[i * 2 + 1]);
+            }
+        }
+
+        ~QuicBuffer()
+        {
+            delete [] Data;
+        }
+    };
+
+    struct QuicTestAddress {
+        QUIC_ADDR Addr;
+        QuicTestAddress(const char* AddrStr, uint16_t Port) {
+            EXPECT_TRUE(QuicAddrFromString(AddrStr, Port, &Addr));
+        }
+    };
+
+    static
+    auto
```

Comment:
`auto` for a function definition? I've never seen that before.

Reply:
I was a little surprised by it too. I couldn't figure out the GTEST type that is returned by the ASSERT macros but apparently the compiler can figure it out.